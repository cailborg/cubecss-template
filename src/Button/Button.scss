.button {
  min-width: 200px;
  &[data-variant="primary"] {
    background: aqua;
    color: purple;
  }
  &[data-variant="secondary"] {
    background: green;
    color: purple;
  }
  &[data-variant="strong"] {
    background: yellow;
    color: purple;
  }
  &[data-variant="default"] {
    background: lightpink;
    color: purple;
  }
  &[data-variant="weak"] {
    background: darkkhaki;
    color: purple;
  }
  &:disabled {
    background: lightgray;
  }
}
