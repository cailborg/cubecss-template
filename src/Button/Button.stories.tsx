import React from "react";
import { Story, Meta } from "@storybook/react/types-6-0";
import { Button, ButtonProps } from "./Button";
import { Icon } from "../Icon/Icon";

export default {
  title: "Components/Button",
  component: Button,
  subcomponents: { Icon },
  argTypes: {
    children: { control: "text" },
  },
} as Meta;

export const Template: Story<ButtonProps> = (args) => <Button {...args}>{args.children}<Icon name ="ArrowRight"/></Button>;
Template.args = { children: 'Button Text' };

