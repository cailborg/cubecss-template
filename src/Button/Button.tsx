import React, { ReactNode } from "react";
import "./Button.scss"

export interface ButtonProps {
  trackingId?: string;
  variant: "primary" | "secondary" | "strong" | "default" | "weak";
  type?: "button" | "submit";
  disabled?: boolean;
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  children: ReactNode;
  ariaLabel?: string;
}

export const Button: React.FC<ButtonProps> = ({
  trackingId,
  variant,
  type,
  disabled,
  onClick,
  ariaLabel,
  children,
}) => {

  return (
    <button
      className={`button text-700 pad-top-300 radius`}
      data-variant={variant}
      disabled={disabled}
      type={type}
      onClick={onClick}
      data-tracking-id={trackingId}
      aria-label={ariaLabel}
    >
      {children}
    </button>
  );
};
