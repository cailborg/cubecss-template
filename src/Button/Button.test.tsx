import React from "react";
import { Button } from "../Button/Button";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<Button variant="primary" size="large">Hello</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
