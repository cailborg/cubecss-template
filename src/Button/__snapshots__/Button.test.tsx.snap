// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="button text-700 pad-top-300 radius"
  data-variant="primary"
>
  Hello
</button>
`;
